### ISSUE TITLE
bug with R 4.0.0 and 4.0.2 containers

### ISSUE THREAD

**Issue by julou** [rocker-org/binder#41]
Not sure where this bug report (jupyterhub/jupyter-rsession-proxy#92) should be posted at best, so [cross-posting](https://github.com/rocker-org/rocker-versioned2/issues/67) here as it seems to be rocker-binder's home.
Sorry if this is bad practice (feel free to close/delete the least appropriate issues in this case) and thank you in advance for your help!

**Comment by cboettig** [comment]
@julou Thanks for the report!  Yup, missed a CMD in the new mapping. My apologies.  Rebuilding the 4.0.x images now and should be up in a little bit.  

p.s. the new 4.0.x images now use the RStudio Package Manager binary builds for Ubuntu Focal (20.04), which should dramatically reduce install times when using binder.  

Again thanks for the report; yeah, any `rocker` based image issues are best filed in these repos just so I see it.  apologies we hadn't caught this!

**Comment by julou** [comment]
Glad that I finally reached the right place ><

I will sound dense here, but could you please instruct how I could force reloading the binder image. Because I now see the 4.0.1 images on dockerhub and all 4.0.x images have an update time that is after your last message so I believe they have been updated but neither https://mybinder.org/v2/gh/rocker-org/binder/master?urlpath=rstudio nor https://mybinder.org/v2/gh/julou/BinderWithR/master?urlpath=rstudio works.

In fact I'm not sure whether it's a refresh issue or that a bug is still lying around: I even updated https://mybinder.org/v2/gh/julou/BinderWithR/master?urlpath=rstudio to force it using 4.0.1 since I know this one is new, but I get the exact same error message…

**Comment by julou** [comment]
> p.s. the new 4.0.x images now use the RStudio Package Manager binary builds for Ubuntu Focal (20.04), which should dramatically reduce install times when using binder.

that's great news!
In fact I already included this feature [for restoring renv with my R-3.6.3 images](https://github.com/julou/BinderWithR/blob/571f0d2f09467cfc32a339112cb2e98ab967e449/.binder/Dockerfile#L26).

**Comment by cboettig** [comment]
Whoopsies, that's what I get for replying to the thread before waiting for the image to finish building.  Had also forgotten to drop out of root and back to USER in the Dockerfile.  It's building again, let's both check again once the new images repost

**Comment by julou** [comment]
Thanks for following up. Still getting the following on both repos:
```
Step 7/9 : RUN chown -R ${NB_USER} ${HOME}
 ---> Running in 2543131df3dd
chown: invalid user: ‘jovyan’
Removing intermediate container 2543131df3dd
The command '/bin/sh -c chown -R ${NB_USER} ${HOME}' returned a non-zero code: 1
```

**Comment by julou** [comment]
Yeah… it now works! Thank you very much :)
Now I have to figure out how to get `renv` installed on 4.0.x images (which works flawlessly in e.g. 3.6 but somehow crashes on 4.0.x). I might open another issue soon :-/

**Comment by julou** [comment]
well… maybe I claimed victory too early!
It turns out that with the current images, the GitHub repo is copied at the root of the file system, instead of in the home directory… Am I right thinking that this must be solved in the Docker image?

**Comment by cboettig** [comment]
Thanks! yeah, you can always address these in your Dockerfiles in the repos, but I think it would be nicer for us to set that in our Dockerfiles for convenience.  Thanks for pointing that out, I'll add a WORKDIR call in now.

**Comment by cboettig** [comment]
I think we should finally be all good here?  At least things look ok if you click the binder button in this repo.  e.g. https://mybinder.org/v2/gh/rocker-org/binder/master?urlpath=rstudio

**Comment by julou** [comment]
hi @cboettig,
I'm sorry for being picky but with the current rocker-org/binder, I get the studio session started in `/home/rstudio` while I see the content of https://github.com/rocker-org/binder at the root of the filesystem. Isn't it [what you intended to fix](#issuecomment-658921138)?
In any case I find it very confusing, I'd expect that one creates a directory with the name of the repo (in this case, probably `/home/binder`, but maybe `/binder` is better for some reason), clone the repo inside, and start the rstudio session from there…
Sorry again if this is not relevant. And thanks a lot for your work on this!
